models/project_board.go

```diff
@@ -0,0 +1,226 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+type (
+	// ProjectBoardType is used to represent a project board type
+	ProjectBoardType uint8
+
+	// ProjectBoardList is a list of all project boards in a repository
+	ProjectBoardList []*ProjectBoard
+)
+
+const (
+	// None is a project board type that has no predefined columns
+	None ProjectBoardType = iota
+
+	// BasicKanban is a project board type that has basic predefined columns
+	BasicKanban
+
+	// BugTriage is a project board type that has predefined columns suited to hunting down bugs
+	BugTriage
+)
+
+// ProjectBoard is used to represent boards on a project
+type ProjectBoard struct {
+	ID      int64 `xorm:"pk autoincr"`
+	Title   string
+	Default bool //if true it collects issues witch are not signed to a specific board jet
+
+	ProjectID int64 `xorm:"INDEX NOT NULL"`
+	RepoID    int64 `xorm:"INDEX NOT NULL"`
```

Comment:
Thanks, and sorry if I'm being a bit of a pain... this one's still not nullable, but it's unclear to me if this actually needs to exist in the database; dropping a couple comments elsewhere to go along these lines. Admittedly, I'm not familiar enough with xorm to know if it's feasible to fetch the RepoID from the project in the couple of places that get ProjectBoard instances from the database.

Reply:
didn't changed this specific code in my refactorys ... I'll have a look at all functions to project & board and minimize the parameters equired